### ISSUE TITLE
Unable to re-import a file-based pool

### ISSUE THREAD

**Issue by Tsaukpaetra** [openzfsonwindows/ZFSin#265]
Using test build OpenZFSOnWindows-debug-20200615, I created a raidz2 pool using blank files using:

    zpool create -O casesensitivity=insensitive -O compression=lz4 -O atime=off -o ashift=12 core raidz2 \\?\C:\disks\core1.bin \\?\C:\disks\core2.bin \\?\C:\disks\core3.bin \\?\C:\disks\core4.bin

I then export it (which succeeds but leaves a dangling drive letter), and then try re-importing it with the command:

    zpool import -d \\?\c:\disks core

Which results in:

```
C:\disks>zpool import -d \\?\c:\disks core
zpool_open_func_win: enter
zpool_open_func_win: enter
zpool_open_func_win: enter
zpool_open_func_win: enter
cannot import 'core': no such pool available
```

There's definitely data there, but whatever probe is being used to detect ZFS pools from files appears to kinda... not.

**Comment by lundman** [comment]
Ah interesting, I'll have to check it out

**Comment by lundman** [comment]
```
C:\src\ZFSin>zpool.exe create BOOM \\?\C:\src\poolfile.bin
C:\src\ZFSin>zpool status
  pool: BOOM
 state: ONLINE
  scan: none requested
config:

        NAME                       STATE     READ WRITE CKSUM
        BOOM                       ONLINE       0     0     0
          \??\C:\src\poolfile.bin  ONLINE       0     0     0

errors: No known data errors

C:\src\ZFSin>
C:\src\ZFSin>zpool export BOOM
zunmount(BOOM,E:\ ) running
zunmount(BOOM,E:\ ) returns 0

C:\src\ZFSin>zpool.exe import -d \\?\C:\src\
cannot open '\\?\C:\src\\*': No error

C:\src\ZFSin>zpool.exe import -d \\?\C:\src
   pool: BOOM
     id: 8541193689867286275
  state: ONLINE
 action: The pool can be imported using its name or numeric identifier.
 config:

        BOOM                       ONLINE
          \??\C:\src\poolfile.bin  ONLINE

C:\src\ZFSin>zpool.exe import -d \\?\C:\src BOOM

C:\src\ZFSin>zpool status
  pool: BOOM
 state: ONLINE
  scan: none requested
config:

        NAME                       STATE     READ WRITE CKSUM
        BOOM                       ONLINE       0     0     0
          \??\C:\src\poolfile.bin  ONLINE       0     0     0

errors: No known data errors

```

**Comment by lundman** [comment]
But, could be with your "leaves a dangling driveletter" has something to do with it, if the export was not clean

**Comment by Tsaukpaetra** [comment]
Yeah, I tried with a new pool, here's my methodology:
https://pastebin.com/qUN6X0Ts
Note that I also tried importing with the `\\?\` prefix and it resulted in the same.
This is one Windows Server 2019 build 17763.615, when I'm bored I'll try on some others, maybe Windows is doing a dumb again...

**Comment by lundman** [comment]
Quite interesting. If I try without "\\?\" it sort of half sees the pool but its unavailable, so it needs the prepend.

Confirm you are running something recent:
```
# zpool -V
zfs-0.2.3-1
zfs-kmod-0.2.3-1
```
that was with today's build.

**Comment by lundman** [comment]
and yep, `-O driveletter=Z` appears not to kick in. But appears to work for non-pool datasets, just not the pool.

**Comment by Tsaukpaetra** [comment]
Ah, yes, the build number does not appear to be in the version string:

```
C:\Windows\system32>zpool -V
zfs-0.2.3-1
zfs-kmod-0.2.3-1
```

But I'm using the 20200615 test build (mentioned in OP).

**Comment by lundman** [comment]
Hmm version string would be better if it had the git rev as well. Must be a way to do that in devstudio

**Comment by Tsaukpaetra** [comment]
FWIW This is the URL I downloaded from: http://www.lundman.net/OpenZFSOnWindows-debug-20200615.exe

**Comment by lundman** [comment]
yep, should work in that too - could be some subtle windows difference, wonder if I still have my windows server VM

**Comment by Tsaukpaetra** [comment]
On the plus side, it handles importing from physical drives (or at least, virtual physical drives) just fine. Same weird issues with exporting though. And I can't get it to mount to the specified drive letter, but that's not as important.
